In [12]:
# I import the necessary libraries
import pandas as pd

**Market Health Index City Dataset**

In [143]:
# I import market health index dataset of cities
df_city = pd.read_csv("MarketHealthIndex_City.csv", encoding="unicode_escape")
df_city.head(2)

RegionType   RegionName         City State  \
0       City     New York     New York    NY   
1       City  Los Angeles  Los Angeles    CA   

                                Metro  \
0                        New York, NY   
1  Los Angeles-Long Beach-Anaheim, CA   

                                 CBSATitle  SizeRank  MarketHealthIndex  \
0  "New York-Newark-Jersey City, NY-NJ-PA"       0.0           9.771856   
1     "Los Angeles-Long Beach-Anaheim, CA"       1.0           7.555211   

   SellForGain  PrevForeclosed  ForeclosureRatio      ZHVI       MoM  \
0        95.73          1.0234            0.0254  693900.0  0.872220   
1        96.37          3.4373            1.0175  633400.0  0.300871   

         YoY  ForecastYoYPctChange  StockOfREOs  NegativeEquity  Delinquency  \
0  14.335146              0.054813          NaN        0.067888     0.099894   
1   7.101792              0.017052          NaN        0.054472     0.031304   

   DaysOnMarket  
0         153.0  
1          66.0

**Market Health Index Metro Dataset**

In [144]:
# I import market health index dataset of metros
df_metro = pd.read_csv("MarketHealthIndex_Metro.csv", encoding="unicode_escape")
df_metro.head(2)

RegionType       RegionName  City State                               Metro  \
0      Metro     New York, NY   NaN    NY                        New York, NY   
1      Metro  Los Angeles, CA   NaN    CA  Los Angeles-Long Beach-Anaheim, CA   

                                 CBSATitle  SizeRank  MarketHealthIndex  \
0  "New York-Newark-Jersey City, NY-NJ-PA"       1.0           5.134707   
1     "Los Angeles-Long Beach-Anaheim, CA"       2.0           7.321712   

   SellForGain  PrevForeclosed  ForeclosureRatio      ZHVI       MoM  \
0        85.92          3.6316            1.7598  426300.0  0.305882   
1        95.17          3.3858            0.9021  613200.0  0.310813   

        YoY  ForecastYoYPctChange  StockOfREOs  NegativeEquity  Delinquency  \
0  7.787611              0.032022          NaN        0.091649     0.125016   
1  5.742369              0.012223          NaN        0.055637     0.030270   

   DaysOnMarket  
0         122.0  
1          63.0

**Market Health Index State Dataset**

In [145]:
# I import market health index dataset of states
df_state = pd.read_csv("MarketHealthIndex_State.csv", encoding="unicode_escape")
df_state.head(2)

RegionType RegionName  City State  Metro  CBSATitle  SizeRank  \
0      State     Alaska   NaN    AK    NaN        NaN       NaN   
1      State    Alabama   NaN    AL    NaN        NaN       NaN   

   MarketHealthIndex  SellForGain  PrevForeclosed  ForeclosureRatio    ZHVI  \
0           3.958333        85.63             NaN               NaN  269100   
1           1.458333        82.25          5.3425            2.3252  126000   

        MoM       YoY  ForecastYoYPctChange  StockOfREOs  NegativeEquity  \
0  0.074377  1.432341              0.045838          NaN        0.146206   
1  0.158983  3.618421              0.029770          NaN        0.142491   

   Delinquency  DaysOnMarket  
0     0.013557          79.0  
1     0.029878          90.0

**Market Health Index Zip Dataset**

In [146]:
# I import market health index dataset of zips
df_zip = pd.read_csv("MarketHealthIndex_Zip.csv", encoding="unicode_escape")
df_zip.head(2)

RegionType  RegionName     City State            Metro          CBSATitle  \
0        Zip        1001   Agawam    MA  Springfield, MA  "Springfield, MA"   
1        Zip        1002  Amherst    MA  Springfield, MA  "Springfield, MA"   

   SizeRank  MarketHealthIndex  SellForGain  PrevForeclosed  ForeclosureRatio  \
0       NaN           2.094542        75.00             NaN            5.7575   
1       NaN           3.099581        92.31             NaN               NaN   

       ZHVI       MoM       YoY  ForecastYoYPctChange  StockOfREOs  \
0  198700.0  0.353535  5.187930              0.028415          NaN   
1  315600.0 -0.221309  3.441495              0.022503          NaN   

   NegativeEquity  Delinquency  DaysOnMarket  
0        0.112505     0.072131          77.5  
1        0.060052     0.050725          93.5

#### **Questions - Answers**

Q1: Which state has the highest market health index?

In [142]:
pd.concat((df_metro,df_zip,df_state,df_city)).groupby('State',as_index=False,sort=False)['MarketHealthIndex'].mean().sort_values("MarketHealthIndex", ascending=False)[:1]

State  MarketHealthIndex
3    TX           8.085605

Q2: Which state has the lowest market health index?

In [147]:
pd.concat((df_metro,df_zip,df_state,df_city)).groupby('State',as_index=False,sort=False)['MarketHealthIndex'].mean().sort_values("MarketHealthIndex", ascending=True)[:1]

State  MarketHealthIndex
41    DE           1.634231

Q3: Which state has the highest delinquency?

In [148]:
pd.concat((df_metro,df_zip,df_state,df_city)).groupby('State',as_index=False,sort=False)['Delinquency'].mean().sort_values("Delinquency", ascending=False)[:1]

State  Delinquency
42    NJ      0.11002

Q4: Which state has the lowest days on the market?

In [149]:
pd.concat((df_metro,df_zip,df_state,df_city)).groupby('State',as_index=False,sort=False)['DaysOnMarket'].mean().sort_values("DaysOnMarket", ascending=True)[:1]

State  DaysOnMarket
3    TX     59.349481

Q5: Which state has the highest percentage of foreclosures?

In [151]:
pd.concat((df_metro,df_zip,df_state,df_city)).groupby('State',as_index=False,sort=False)['ForeclosureRatio'].mean().sort_values("ForeclosureRatio", ascending=False)[:1]

State  ForeclosureRatio
35    ID          295.9091

Q6: Which state has the lowest percentage of foreclosures?

In [152]:
pd.concat((df_metro,df_zip,df_state,df_city)).groupby('State',as_index=False,sort=False)['ForeclosureRatio'].mean().sort_values("ForeclosureRatio", ascending=True)[:1]

State  ForeclosureRatio
6    FL            0.3697

**Summary Table**

|                             | State | Value |
| --------------------------- | ----- | ----- |
| Highest Market Health Index | TX    | 8.085  |
| Lowest Market Health Index  | DE    | 1.634  |
| Highest Delinquency         | NJ    | 0.110  |
| Lowest Days on Market       | TX    | 59.349 |
| Highest Foreclosure Ratio   | ID    | 295.909    |
| Lowest Foreclosure Ratio    | FL    | 0.369  |